mne/forward/_make_forward.py

```diff
@@ -288,17 +293,29 @@ def _prep_meg_channels(info, accurate=True, exclude=(), ignore_ref=False,
     picks = pick_types(info, meg=True, ref_meg=ref_meg, exclude=exclude)
     meg_info = pick_info(info, picks) if nmeg > 0 else None
 
-    # Create coil descriptions with transformation to head or MRI frame
+    # Create coil descriptions with transformation to head or device frame
     templates = _read_coil_defs(elekta_defs=elekta_defs)
 
-    megcoils = _create_meg_coils(megchs, accuracy, info['dev_head_t'],
-                                 templates)
+    if head_frame:
+        _print_coord_trans(info['dev_head_t'])
+        transform = info['dev_head_t']
+    else:
+        transform = None
+
+    megcoils = _create_meg_coils(megchs, accuracy, transform, templates)
+
     if ncomp > 0:
         logger.info('%d compensation data sets in %s' % (ncomp_data,
                                                          info_extra))
-        compcoils = _create_meg_coils(compchs, 'normal', info['dev_head_t'],
-                                      templates)
-    logger.info('Head coordinate MEG coil definitions created.')
+        compcoils = _create_meg_coils(compchs, 'normal', transform, templates)
+
+    # Check that coordinate frame is correct and log it
+    if head_frame:
+        assert megcoils[0]['coord_frame'] == FIFF.FIFFV_COORD_HEAD
+        logger.info('MEG coil definitions created in head coordinates.')
+    else:
+        assert megcoils[0]['coord_frame'] == FIFF.FIFFV_COORD_DEVICE
```

Comment:
I'm surprised this assertion passes since I thought in latest `master` the coord frame was always `head` (`transform=None` is assumed to be identity transform)